### PR TITLE
Port stocks to fn3 and introduce an App component.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth 

Known bugs:
- snackbar doesn't shift the fab when it comes up. I plan to fix that next week by making the scaffold put those two components in the same column anchored to the bottom.
- tapping on a toggleable doesn't seem to work.
- changing theme doesn't animate the theme, because AnimatedComponent is missing.

File: sky/packages/sky/lib/src/fn3/app.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  final Map<String, RouteBuilder> routes;
+
+  AppState createState() => new AppState();
+}
+
+class AppState extends State<App> {
+
+  GlobalObjectKey _navigator;
+
+  void initState(BuildContext context) {
```
Comment (abarth) [inline sky/packages/sky/lib/src/fn3/app.dart]: We should remove the `context` parameter to this function now that there's a `context` getter on `State`.

**Comment by abarth** [comment]
LGTM

File: sky/packages/sky/lib/src/fn3/app.dart (hunk 1 above)
Comment (Hixie) [inline sky/packages/sky/lib/src/fn3/app.dart]: Filed #1364.

**Pull request merged**